### ISSUE TITLE
Evaluation crash when watched metric matches phase prefix due to mixed figure and scalar payload in aggregate_and_plot_metrics

### ISSUE THREAD

**Issue by kshirajahere** [mllam/neural-lam#608]
### Summary

When a watched metric actually matches the current phase prefix (for example `val_rmse` during validation or `test_rmse` during test) and `--var_leads_metrics_watch` is non-empty, evaluation can crash inside `ARModel.aggregate_and_plot_metrics()`.

### Root cause

`create_metric_log_dict()` currently mixes two different payload types into the same `log_dict`:

- figure objects for heatmap plots
- scalar tensor values for exact watched metric values

Relevant lines:
- [neural_lam/models/ar_model.py:693-699 ](https://github.com/mllam/neural-lam/blob/main/neural_lam/models/ar_model.py#L693-L699)
- [neural_lam/models/ar_model.py:737-754](https://github.com/mllam/neural-lam/blob/main/neural_lam/models/ar_model.py#L737-L754)

Specifically:
- line 699 inserts `metric_tensor[step - 1, var_i]` into `log_dict`
- line 737 immediately asserts that **all** `log_dict` values are `plt.Figure`
- line 754 then tries to send every value through `logger.log_image(...)`

So a matching watched metric does not just fail the assertion; the data structure itself is incompatible with the downstream logging path.

### Reproduction

A minimal direct reproduction is:

```python
import torch
import matplotlib.pyplot as plt
from types import SimpleNamespace
from unittest.mock import patch
from neural_lam.models.ar_model import ARModel

class StubDatastore:
    def get_vars_names(self, category):
        return ["var0", "var1"]

class StubModule:
    pass

module = StubModule()
module.args = SimpleNamespace(
    metrics_watch=["val_rmse"],
    var_leads_metrics_watch={0: [1]},
)
module.matched_metrics = set()
module._datastore = StubDatastore()
module.state_std = torch.ones(2)
module.trainer = SimpleNamespace(is_global_zero=True, sanity_checking=False)
module.all_gather_cat = lambda tensor: tensor
module.create_metric_log_dict = ARModel.create_metric_log_dict.__get__(module, StubModule)
module.aggregate_and_plot_metrics = ARModel.aggregate_and_plot_metrics.__get__(module, StubModule)

metrics_dict = {"mse": [torch.ones(1, 2, 2)]}

with patch("neural_lam.models.ar_model.vis.plot_error_map", return_value=plt.figure()):
    module.aggregate_and_plot_metrics(metrics_dict, prefix="val")

**Comment by sadamov** [comment]
is this not being solved in https://github.com/mllam/neural-lam/pull/303?

**Comment by kshirajahere** [comment]
@sadamov Thats exactly what i was thinking that I feel like this was raised before. And just came to close out this issue 😓. My bad should have checked before raising
